x-pack/plugins/alerting/README.md

```diff
@@ -0,0 +1,227 @@
+|---|---|---|
+|id|Unique identifier for the alert type. For convention purposes, ids starting with `.` are reserved for built in alert types. We recommend using a convention like `<plugin_id>.mySpecialAlert` for your alert types to avoid conflicting with another plugin.|string|
+|name|A user friendly name for the alert type. These will be displayed in dropdowns when chosing alert types.|string|
+|validate.params|When developing an alert type, you can choose to accept a series of parameters. You may also have the parameters validated before they are passed to the `execute` function or created as an alert saved object. In order to do this, provide a joi schema that we will use to validate the `params` attribute.|Joi schema|
+|execute|This is where the code of the alert type lives. This is a function to be called when executing an alert on an interval basis. For full details, see executor section below.|Function|
+
+### Executor
+
+This is the primary function for an alert type, whenever the alert needs to execute, this function will perform the execution. It receives a variety of parameters, the following table describes the properties the executor receives.
+
+**execute(options)**
+
+|Property|Description|
+|---|---|
+|services.callCluster(path, opts)|Use this to do elasticsearch queries on the cluster Kibana connects to. This function is the same as any other `callCluster` in Kibana.<br><br>**NOTE**: This currently authenticates as the Kibana internal user, this will change in a future PR.|
+|services.savedObjectsClient|This is an instance of the saved objects client. This provides the ability to do CRUD on any saved objects within the same space the alert lives in.<br><br>**NOTE**: This currently only works when security is disabled. A future PR will add support for enabled security using Elasticsearch API tokens.|
+|services.log(tags, [data], [timestamp])|Use this to create server logs. (This is the same function as server.log)|
+|scheduledRunAt|The date and time the alert type execution was scheduled to be called.|
+|previousScheduledRunAt|The previous date and time the alert type was scheduled to be called.|
+|params|Parameters for the execution. This is where the parameters you require will be passed in. (example threshold). Use alert type validation to ensure values are set before execution.|
+|state|State returned from previous execution. This is the alert level state, what is returned by the executor will be serialized and provided here at the next execution.|
+
+### Example
+
+```
+server.plugins.alerting.registerType({
+  id: 'my-alert-type',
+  name: 'My alert type',
+  validate: {
+    params: Joi.object()
+      .keys({
+        myParam: Joi.string().required(),
+      })
+      .required(),
+  },
+  async execute({
+    scheduledRunAt,
+    previousScheduledRunAt,
+    services,
+    params,
+    state,
+  }: AlertExecuteOptions) {  
+    // Use this example to fire a single action
+    alertInstanceFactory('server_1')
```

Comment:
Also, I'm not clear on what the string you pass to alert instance factory does.  This needs to be explained I think.